src/gamma_inc.jl

```diff
@@ -421,39 +421,32 @@ See also: [`gamma_inc(a,x,ind)`](@ref SpecialFunctions.gamma_inc)
 """
 function gamma_inc_taylor(a::Float64, x::Float64, ind::Integer)
     acc = acc0[ind + 1]
-    wk = zeros(30)
-    flag = false
-    apn = a + 1.0
-    t = x/apn
-    wk[1] = t
-    loop = 2
-    for indx = 2:20
-       apn += 1.0
-       t *= x/apn
-       if t <= 1.0e-3
-           loop = indx
-           flag = true
-           break
-       end
-       wk[indx] = t
-    end
-    if !flag
-        loop = 20
+    wk = zeros(20)
+    apn = a
+
+    # compute and store larger terms in wk, to add from small to large
+    t = 1
+    i = 0
+    while i < 20
+        i += 1
+        apn += 1.0
+        t *= x / apn
+        t > 1.0e-3 || break
+        wk[i] = t
     end
+
+    # sum the smaller terms directly
     sm = t
-    tol = 0.5*acc #tolerance
-    while true
-       apn += 1.0
-       t *= x/apn
-       sm += t
-       if t <= tol
-           break
-       end
-    end
-    for j = loop-1:-1:1
-       sm += wk[j]
+    tolerance = 0.5acc
+    while t > tolerance
+        apn += 1.0
+        t *= x / apn
+        sm += t
     end
-    p = (rgammax(a, x)/a)*(1.0 + sm)
+
+    # Sum terms from small to large
+    sm += sum(reverse!(@view wk[1:i]))
```

Comment:
If you want a specific order, then I guess you shouldn't use `sum` which uses pairwise summation. Using views and `reverse!` seems also a bit complex for this task. So maybe just iterate over the array in the desired order?
```suggestion
    # Sum terms from small to large
    for j in i:(-1):1
        sm += wk[j]
    end
```

Reply:
Pairwise, as in `a + b + c + d = ab + cd = abcd` ? This isn't documented AFAICT.

Reply:
https://en.wikipedia.org/wiki/Pairwise_summation

Reply:
Ah. I suppose to be fair, the tests all passed using it, so it seems to work fine. But I removed it anyway.